### PR TITLE
Added several bridge styles

### LINKED ISSUE #4400: Rendering of an outline for pedestrian / bike bridges
**Is your feature request related to a problem? Please describe.**

I would like to see the rendering of an outline for pedestrian / bike bridges, in a similar way to car bridges (and to what the standard OSM rendering is doing). It concerns paths with tags:

| bridge     | yes  |
| highway  | path |

Currently there is no indication there is a bridge: It would be useful for smaller bridges, where it is unclear if there is indeed a bridge or if it is a ford (there is a tiny dot rendered for fords but the data is not very consistent and it is often missing).

**Describe the solution you would like**

An outline similar to the rendering of car bridges is drawn.

**Describe alternatives you have considered**

For example:

OM
Screenshot_20230205_124047_Organic Maps

OSM

(from https://www.openstreetmap.org/#map=18/45.97650/6.30859)

**Additional context**

There are somewhat related issues with #3023 and #3322 but they are more about rendering a bridge area (for large bridges).

### PR DESCRIPTION
**Pull request by map-per**
Closes #4400

Added bridges for footway, cycleway, path, steps, track and bridleway.

(last updated: 2023-08-26)

| before | now | OSM |
|---|---|---|
|  Screenshot_2023-08-24-15-33-39-313_app organicmaps | Screenshot_2023-08-26-15-04-44-837_app organicmaps beta | [OSM](https://www.openstreetmap.org/way/159542727)|
| Screenshot_2023-08-24-15-34-43-988_app organicmaps| Screenshot_2023-08-26-15-05-10-565_app organicmaps beta | [OSM](https://www.openstreetmap.org/way/175099459) |

Screenshot_2023-08-26-15-06-01-599_app organicmaps beta
Footway: https://www.openstreetmap.org/way/93093861
Cycleway: https://www.openstreetmap.org/way/19687050
Paths: https://www.openstreetmap.org/way/34213185, https://www.openstreetmap.org/way/19688542
Screenshot_2023-08-26-15-06-06-471_app organicmaps beta
Screenshot_2023-08-26-15-06-41-260_app organicmaps beta
Screenshot_2023-08-26-15-06-51-622_app organicmaps beta
Screenshot_2023-08-26-15-07-00-504_app organicmaps beta
Screenshot_2023-08-26-15-07-09-801_app organicmaps beta
Screenshot_2023-08-26-15-07-16-789_app organicmaps beta
Screenshot_2023-08-26-15-07-21-968_app organicmaps beta
Screenshot_2023-08-26-15-07-31-258_app organicmaps beta
Screenshot_2023-08-26-15-07-38-733_app organicmaps beta

Screenshot_2023-08-26-15-08-17-458_app organicmaps beta
Steps: https://www.openstreetmap.org/way/759824860
Screenshot_2023-08-26-15-08-23-236_app organicmaps beta
Screenshot_2023-08-26-15-08-29-978_app organicmaps beta
Screenshot_2023-08-26-15-08-34-519_app organicmaps beta
Screenshot_2023-08-26-15-08-39-613_app organicmaps beta
Screenshot_2023-08-26-15-08-49-403_app organicmaps beta

Screenshot_2023-08-26-15-09-11-643_app organicmaps beta
Track: https://www.openstreetmap.org/way/37857500

Screenshot_2023-08-26-15-09-44-581_app organicmaps beta
Bridleway: https://www.openstreetmap.org/way/37809636

### CODE REVIEW

File: data/styles/clear/include/Roads.mapcss [hunk 1]
```diff
@@ -659,31 +667,17 @@ line|z16-[highway=pedestrian][tunnel?]
...
-line|z15-[highway=steps][bridge?]::bridgeblack,
-*/
 line|z16-[highway=road][bridge?]::bridgeblack,
 line|z16-[highway=service][bridge?]::bridgeblack
 {casing-width-add: 0.2;}
-line|z16-[highway=pedestrian][bridge?]::bridgeblack,
-/* TODO: test this style before adding
-line|z16-[highway=footway][bridge?]::bridgeblack,
-*/
+line|z16-[highway=pedestrian][bridge?]::bridgeblack
```
Comment (pastk) [inline data/styles/clear/include/Roads.mapcss:680]: nit: its better to keep a trailing comma everywhere, this way its less diff noise with subsequent additions

File: data/styles/clear/include/Roads.mapcss [hunk 2]
```diff
@@ -787,6 +794,16 @@ line|z19-[route=hiking],
 line|z19-[highway=footway][sac_scale],
 {width: 4; dashes: 8,4.5;}
 
+line|z16-[highway=path][bridge?]::bridgewhite
+{casing-width-add: 0.5;}
+line|z16-[highway=path][bridge?]::bridgeblack
```
Comment (pastk) [inline data/styles/clear/include/Roads.mapcss:799]: for railways bridges we usually display ::bridgewhite only first , e.g. at z13-
and then add ::bridgeblack later at e.g. z16-

I think it works very well and makes bridges visually "ligher weight" at smaller zooms.
Maybe try it with paths/cycleways/etc. also? I.e. add a ::bridgeblack staring at z17- or z18- even.

File: data/styles/clear/include/Roads.mapcss [hunk 3]
```diff
@@ -814,6 +836,23 @@ line|z18[highway=steps]
...
+line|z19-[highway=steps][bridge?]::bridgewhite
+{casing-width-add: -3.5;}
+line|z16[highway=steps][bridge?]::bridgeblack
+{casing-width-add: -1.5;}
+line|z17[highway=steps][bridge?]::bridgeblack
+{casing-width-add: -2;}
+line|z18[highway=steps][bridge?]::bridgeblack
+{casing-width-add: -2.5;}
+line|z19-[highway=steps][bridge?]::bridgeblack
+{casing-width-add: -3;}
```
Comment (pastk) [inline data/styles/clear/include/Roads.mapcss:854]: do stairs actually fit into the bridge outline with these params? :)
would be great to see a screenshot!

File: data/styles/clear/include/priorities_3_FG.prio.txt
Comment (pastk) [inline data/styles/clear/include/priorities_3_FG.prio.txt:1]: please add ::bridgeblack and ::bridgewhite lines into this file to set priorities for them;
the main line should be higher than ::bridgewhite which should be higher than ::bridgeblack (e.g. see how its done for railways or highways)

**Review by pastk — CHANGES_REQUESTED** [review]
Thanks a lot for this PR! We missed those bridges for a long time!

File: data/styles/clear/include/Roads.mapcss (hunk 3 above)
Comment (map-per) [reply data/styles/clear/include/Roads.mapcss:854]: There are screenshots of https://www.openstreetmap.org/way/759824860 in the first post

File: data/resources-6plus_clear/symbols.sdf
[the remark this answers is not in the document]
Comment (map-per) [reply data/resources-6plus_clear/symbols.sdf:1]: resolved

File: data/styles/clear/include/Roads.mapcss (hunk 1 above)
Comment (map-per) [reply data/styles/clear/include/Roads.mapcss]: resolved

File: data/styles/clear/include/Roads.mapcss (hunk 2 above)
Comment (map-per) [reply data/styles/clear/include/Roads.mapcss]: Changed ::bridgewhite to start at z13 and ::bridgeblack at z17

File: data/styles/clear/include/priorities_3_FG.prio.txt
Comment (map-per) [reply data/styles/clear/include/priorities_3_FG.prio.txt:1]: Thanks! Added them

**Comment by map-per** [comment]
Thanks for the rewiew! I resolved all comments. It's really nice that you get feedback so quickly in OM. In CartoOSM I often have to wait for months

File: data/styles/clear/include/Roads.mapcss (hunk 2 above)
Comment (pastk) [reply data/styles/clear/include/Roads.mapcss]: ::bridgeblack at z17- sounds good!

Bbut ::bridgewhite at z13 is a bit too early, sorry :)

E.g. highway=path is visbible from z14- and highway=footway from z15-, so it doesn't make sense to display ::bridgewhite before the main line.

I suggest make ::bridgewhite start from z15-.

**Comment by pastk** [comment]
I think https://github.com/organicmaps/organicmaps/pull/5808#discussion_r1306435642 is the only thing left to be fixed.

File: data/styles/clear/include/Roads.mapcss (hunk 2 above)
Comment (map-per) [reply data/styles/clear/include/Roads.mapcss]: 👍 That makes sense, changed it to z15

**Review by pastk — APPROVED** [review]

**Review by vng — APPROVED** [review]

**Comment by map-per** [comment]
👍 That was quick, will do cycleway, track and dridleway tunnels next

**Pull request merged**